internal/destregistry/providers/destrabbitmq/destrabbitmq_publish_test.go

```diff
@@ -1,188 +1,138 @@
-	return c.messages
+// Make sure MockChannel implements the needed methods
+func (m *MockChannel) Close() error {
+	return nil // Simple implementation for testing
 }
 
-func (c *RabbitMQConsumer) Close() error {
-	if c.channel != nil {
-		c.channel.Close()
-	}
-	if c.conn != nil {
-		c.conn.Close()
-	}
-	close(c.messages)
-	return nil
+func (m *MockChannel) IsClosed() bool {
+	return false // Simple implementation for testing
 }
 
-// RabbitMQAsserter implements provider-specific message assertions
-type RabbitMQAsserter struct{}
-
-func (a *RabbitMQAsserter) AssertMessage(t testsuite.TestingT, msg testsuite.Message, event models.Event) {
-	delivery, ok := msg.Raw.(amqp091.Delivery)
-	assert.True(t, ok, "raw message should be amqp.Delivery")
-
-	// Assert RabbitMQ-specific properties
-	assert.Equal(t, "application/json", delivery.ContentType)
-	// assert.NotEmpty(t, delivery.MessageId)
-	// assert.NotEmpty(t, delivery.Timestamp)
-
-	// Could add more RabbitMQ-specific assertions:
-	// - Exchange routing
-	// - Message persistence
-	// - Priority
-	// - etc.
+// MockConnection mocks the Connection interface for testing
+type MockConnection struct {
+	mock.Mock
 }
 
-// RabbitMQPublishSuite reimplements the publish tests using the shared test suite
-type RabbitMQPublishSuite struct {
-	testsuite.PublisherSuite
-	consumer *RabbitMQConsumer
+func (m *MockConnection) Close() error {
+	args := m.Called()
+	return args.Error(0)
 }
 
-func (s *RabbitMQPublishSuite) SetupSuite() {
-	t := s.T()
-	t.Cleanup(testinfra.Start(t))
-	mqConfig := testinfra.NewMQRabbitMQConfig(t)
-
-	provider, err := destrabbitmq.New(testutil.Registry.MetadataLoader())
-	require.NoError(t, err)
-
-	dest := testutil.DestinationFactory.Any(
-		testutil.DestinationFactory.WithType("rabbitmq"),
-		testutil.DestinationFactory.WithConfig(map[string]string{
-			"server_url":  testutil.ExtractRabbitURL(mqConfig.RabbitMQ.ServerURL),
-			"exchange":    mqConfig.RabbitMQ.Exchange,
-			"routing_key": mqConfig.RabbitMQ.Queue,
-			// "tls":         "false", // should default to false if omitted
-		}),
-		testutil.DestinationFactory.WithCredentials(map[string]string{
-			"username": testutil.ExtractRabbitUsername(mqConfig.RabbitMQ.ServerURL),
-			"password": testutil.ExtractRabbitPassword(mqConfig.RabbitMQ.ServerURL),
-		}),
-	)
-
-	consumer, err := NewRabbitMQConsumer(mqConfig)
-	require.NoError(t, err)
-	s.consumer = consumer
-
-	s.InitSuite(testsuite.Config{
-		Provider: provider,
-		Dest:     &dest,
-		Consumer: consumer,
-		Asserter: &RabbitMQAsserter{}, // Add RabbitMQ-specific assertions
-	})
+func (m *MockConnection) IsClosed() bool {
+	args := m.Called()
+	return args.Bool(0)
 }
 
-func (s *RabbitMQPublishSuite) TearDownSuite() {
-	if s.consumer != nil {
-		s.consumer.Close()
-	}
-}
-
-func TestRabbitMQPublish(t *testing.T) {
-	if testing.Short() {
-		t.Skip("Skipping integration test in short mode")
-	}
-	suite.Run(t, new(RabbitMQPublishSuite))
-}
```

Comment:
This is the code that initiate running the integration test for this destination as I described above.

@leggetter

Reply:
Yeah, it's moved into another file.

Reply:
https://github.com/hookdeck/outpost/pull/285/files#diff-7cdc978e39f75b93a912d427fb1862ed46b3913ab3fb5d339f82b5aff6199d8fR170

Reply:
@leggetter I see, sorry my bad I somehow missed that.

Anyway, if you don't mind, I removed the unit test you set up in favor of the original integration test for consistency purposes. I updated that test to work with the new routing key convention, so please feel free to review if you're interested in how it works.

Do let me know if you feel the unit test is helpful moving forward, we can certainly revert and include that if that helps your workflow. I'm also around if you'd like to connect and discuss the workflow further and how to run the integration test suite.